### DISCUSSION TITLE
Linhas do tempo interativas com python

### DISCUSSION THREAD

**Discussion by gbrlbrt** [clube-de-leitura-python/pense-em-python-2021#22] [Conversa geral]
Oi, gente, tudo certo?

Como nossos encontros são bem espaçados, eu tava querendo começar algum projeto pessoal que tivesse um resultado concreto para que eu treine mais o python e que eu possa ir melhorando com o tempo. A minha ideia é fazer algo relacionado à minha pesquisa do doutorado: eu estudo um conjunto de autores do período de Weimar que estavam tentando entender a relação entre direito, democracia e capitalismo.
A minha ideia era fazer uma linha do tempo que, além dos principais eventos históricos, marcasse também o momento de publicação de cada texto e que fosse interativo no sentido de que clicando em cada texto ele te apresentasse de uma maneira bem visual as referencias cruzadas entre as obras (citações, contra quem estava escrevendo, etc) - inicialmente talvez uma caixa de texto e depois eu vou complexificando.

Vocês teriam alguma recomendação de por onde começo a aprender a fazer isso? Não é pra ser nada muito complexo ou automatizado de início, mas queria já começar em alguma coisa que permitisse uma apresentação visualmente bem interessante mais pra frente.

Valeu!

**Comment by villares** [comment]
Puxa, que ideia legal!  Olha, pra fazer linhas do tempo interativas, em geral o pessoal vai preferir fazer algo online, e por isso usar JavaScript... mas esse lance de visualização dá sim pra fazer com Python.

A ferramenta que eu mais uso é uma mistura de Python com um vocabulário gráfico que é o Processing. Acho que é uma ideia muito legal e desafiador. Para fazer interativo não vai ser trivial, mas é bacana. Vou dar uma pensada no que te indicar.  
Palpite: faz uma planilha com as obras que eu te mostro como ler a planilha no Python pra dar um pontapé inicial :)

Se você quiser olhar uma ferramenta existente de time-lines... tem esta aqui: https://time.graphics/pt/
E o Zotero.org também cria uma a partir de referências bibliográficas (mas nunca usei essa feature)

**Reply by gbrlbrt** [reply]
Opa, obrigado!

Então, pelo que eu vi, preciso começar a aprender como armazenar todos os textos (citações, refs, etc.),  talvez um GUI que edita eles.

A minha ideia era construir isso no python mesmo e depois, como eu não acho que vou aprender java tbm, montar um desses site pessoal que você mal precisa saber programação pra fazer e só colocar ele lá, quase como um app.
Faz sentido isso?

E sim, queria usar o Zotero como base de dados mesmo. Mas tava tentando descobrir a parte funcional antes e só depois ir pora visual.

**Reply by victorpmor** [reply]
Cara... já que no final das contas vai ser uma espécie site,  então me passou pela cabeça a possibilidade de usar Django - que é um framework do python por meio do qual se faz sites (utilizando a linguagem python!).
Eu usei muito pouco. Mas é uma ferramente bastaaaaaaante simples de se fazer site. Mas bastante de verdade mesmo! Porque, no seu caso, seria um site dinâmico, portanto, usaria banco de dados para armazenar e resgatar os dados. Por meio do django vc faz tudo isso somente com a linguagem python - nao precisaria aprender PHP, por exemplo, para conversar com o banco de dados. Bom... teria que aprender um vocabulário a mais, né... 

Por exemplo, pra 'montar' um banco de dados, vc precisaria, praticamente, só digitar algo como isso:

from django.db import models

class Imovel(models.Model):
    titulo = models.CharField('Titulo', max_length=200, blank=False)
    slug = models.SlugField('Slug')
    cep = models.IntegerField('CEP', blank=False)
    estado = models.CharField('Estado', max_length=50, blank=False)
    cidade = models.CharField('Cidade', max_length=50, blank=False)
    bairro = models.CharField('Bairro', max_length=200, blank=False)
    fotos = models.ImageField('Fotos',
                              upload_to='imagens/%Y/')

**Reply by gbrlbrt** [reply]
Legal, muito obrigado!

**Comment by claudewick** [comment]
Vai ser uma app bem visual, também acho que web seria a melhor opção. Não precisa usar Javascript, você pode usar o framework Flask que é em Python.
